cola/jax_fns.py

```diff
@@ -121,7 +121,7 @@ def lu_solve(a, b):
 
 
 def get_device(array):
-    if not isinstance(array, jax.core.Tracer):
+    if not isinstance(array, jax.core.Tracer) and hasattr(array, 'device'):
```

Comment:
I added this second check because, at least in the tests, the Jax xnp seems to be used with numpy arrays which do not have a device attribute.